lampod/src/ln/onion_message.rs

```diff
@@ -96,9 +97,10 @@ where
         let first_node = match destination {
             Destination::Node(node_id) => node_id,
             Destination::BlindedPath(BlindedPath {
-                introduction_node_id: node_id,
+                introduction_node: IntroductionNode::NodeId(node_id),
                 ..
             }) => node_id,
+            _ => panic!("Invalid destination"),
```

Comment:
we should return the error, lampo never panic

Reply:
```
diff --git a/lampod/src/ln/onion_message.rs b/lampod/src/ln/onion_message.rs
index 3ce49c6..dcf553c 100644
--- a/lampod/src/ln/onion_message.rs
+++ b/lampod/src/ln/onion_message.rs
@@ -95,13 +95,13 @@ where
         destination: lampo_common::ldk::onion_message::messenger::Destination,
     ) -> Result<lampo_common::ldk::onion_message::messenger::OnionMessagePath, ()> {
         let first_node = match destination {
-            Destination::Node(node_id) => node_id,
+            Destination::Node(node_id) => Ok(node_id),
             Destination::BlindedPath(BlindedPath {
                 introduction_node: IntroductionNode::NodeId(node_id),
                 ..
-            }) => node_id,
-            _ => panic!("Invalid destination"),
-        };
+            }) => Ok(node_id),
+            _ => Err("Invalid destination".to_string()),
+        }.unwrap();
```

Is this okay?

Reply:
I do not know, maybe we should support the destination, why fo you it is an invalid destination?